monai/data/utils.py

```diff
@@ -536,3 +537,26 @@ def is_supported_format(filename: Union[Sequence[str], str], suffixes: Sequence[
             break
 
     return supported_format
+
+
+def partition_dataset(data: Sequence[Any], shuffle: bool = False, seed: int = 0):
```

Comment:
I think this function is not generic enough to be a part of the core APIs, could you move it to the example file for now? also shall we pass the rank information explicitly to this function instead of relying on the sampler's default behaviour of distributed groups? (mainly for better readability)

Reply:
Hi @wyli ,

Thanks for your review.
May I know why you think it's not generic? Basically, it's based on the `DatasetPartitioner` in PyTorch official docs: https://pytorch.org/tutorials/intermediate/dist_tuto.html#distributed-training
And I can copy the logic of `DistributedSampler` instead of calling it to make it more clear if you guys feel it's better?
@ericspod What do you think? Could you please also share some comments?

Thanks.

Reply:
The function is doing something quite simple, ie. return a set of values from the dataset by selecting a partition set of indices. I would copy the logic from the class in a condensed form so that you're not creating an object just to use one method then toss it away.  We might want to change the function to accept a Numpy RandState object instead of using the boolean flag so that we can have a deterministic shuffling, if the state is None then no shuffling.

Reply:
Hi @ericspod and @wyli ,

Thanks for your suggestions.
I updated the PR according to your comments.
Could you please help review it again?
I didn't add `numpy randstate` because we should always set random seed and use a deterministic shuffling, otherwise, every process may have different shuffle results.

Thanks.